sdk/internal/recording/recording.go

```diff
@@ -814,7 +801,76 @@ func Stop(t *testing.T, options *RecordingOptions) error {
 	return err
 }
 
-// This looks up an environment variable and if it is not found, returns the recordedValue
+func getRandomSource(t *testing.T) rand.Source {
+	if testStruct, ok := testSuite.Load(t.Name()); ok {
+		if testStruct.recordingRandomSource != nil {
+			return testStruct.recordingRandomSource
+		}
+	}
+
+	var seed int64
+	var err error
+
+	variables := GetVariables(t)
+	seedString, ok := variables[recordingRandomSeedVariableName]
+	if ok {
+		seed, err = strconv.ParseInt(seedString.(string), 10, 64)
+	}
+
+	// We did not have a random seed already stored; create a new one
+	if !ok || err != nil || GetRecordMode() == "live" {
+		seed = time.Now().Unix()
+	}
+
+	source := rand.NewSource(seed)
+	if testStruct, ok := testSuite.Load(t.Name()); ok {
+		testStruct.recordingSeed = seed
+		testStruct.recordingRandomSource = source
+		testSuite.Store(t.Name(), testStruct)
+	}
+
+	return source
+}
+
+// GenerateAlphaNumericID will generate a recorded random alpha numeric id.
+// When playback mode, a new random seed will be used, otherwise, the random seed is stable and will be stored in recording file.
```

Comment:
I think this comment is wrong and should be the same as `Recording.GenerateAlphaNumericID()` yes?